pkgdb2client/admin.py

```diff
@@ -0,0 +1,445 @@
+            user=action['user'],
+        )
+
+        PKGDBCLIENT.handle_api_call(
+            '/admin/action/status',
+            data={
+                'id': actionid,
+                'status': 'Approved'
+            }
+        )
+
+    elif decision in ('deny', 'd'):
+        message = raw_input(
+            'Could you explain why you declined this request? (this message '
+            'will be sent to the user)\n=>')
+        data = PKGDBCLIENT.handle_api_call(
+            '/admin/action/status',
+            data={
+                'id': actionid,
+                'status': 'Denied',
+                'message': message,
+            }
+        )
+
+    else:
+        data = {
+            'messages': ['Action {0} un-touched'.format(actionid)]
+        }
+
+
+    return data
+
+
+def do_process(args):
+    ''' Process a specific admin action.
+
+    '''
+    LOG.info("user   : {0}".format(args.username))
+
+    for actionid in args.actionid:
+        LOG.info("action : {0}".format(actionid))
+        action = PKGDBCLIENT.handle_api_call('/admin/action/%s' % actionid)
+
+        print _action2msg(action)
+
+        if action['status'] != 'Awaiting Review':
+            print 'Action #%s is not Awaiting Review - Current status: %s' % (
+                action['id'], action['status'])
+            return
+
+        if action['action'] == 'request.package':
+            try:
+                PKGDBCLIENT.get_package(action['info']['pkg_name'])
+                print 'Package {0} found, requalifying request.package ' \
+                    'in request.branch'.format(action['info']['pkg_name'])
+                # Adjusting the input format
+                action['action'] = 'request.branch'
+                action['package'] = {'name': action['info']['pkg_name']}
+                action['collection'] = {
+                    'branchname': action['info']['pkg_collection']}
+
+            except pkgdb2client.PkgDBException:
+                pass
+
+        if action['action'] == 'request.package':
+            data = __handle_request_package(actionid, action)
+        elif action['action'] == 'request.branch':
+            data = __handle_request_branche(actionid, action)
+        else:
+            print 'Action %s not supported by pkgdb-cli' % action['action']
+
+        for msg in data.get('messages', []):
+            print msg
+        print ''
+
+
+def main():
+    ''' Main function '''
+    # Set up parser for global args
+    parser = setup_parser()
+    # Parse the commandline
+    try:
+        arg = parser.parse_args()
+    except argparse.ArgumentTypeError, err:
+        print "\nError: {0}".format(err)
+        return 2
+
+    logging.basicConfig()
+    if arg.debug:
+        LOG.setLevel(logging.DEBUG)
+        PKGDBLOG.setLevel(logging.DEBUG)
+    elif arg.verbose:
+        LOG.setLevel(logging.INFO)
+
+    if arg.test:
+        global PKGDBCLIENT
+        print "Testing environment"
+        PKGDBCLIENT = PkgDB(
```

Comment:
Perhaps make this configurable by adding ``--insecure`` and ``--baseurl`` options instead of ``--test``.  That would be more flexible.  You could point it at staging, or localhost, or production (by default).  Etc.

That will be nice for using it in staging releng processes some day.